setup.py

```diff
@@ -1,80 +1,215 @@
+  Unpack package file to the specified directory
   """
-  packages = []
-  for root, _, files in os.walk(repositoryDir + "/nupic"):
-    if "__init__.py" in files:
-      subdir = root.replace(repositoryDir + "/", "")
-      packages.append(subdir.replace("/", "."))
-  return packages
+
+  if not silent:
+    print "Unpacking %s into %s..." % (package, destDir)
+
+  with tarfile.open(package, "r:gz") as tarFileObj:
+    tarFileObj.extractall(destDir)
+
+  # Copy subdirectories to a level up
+  subDirs = os.listdir(destDir + "/" + dirToUnpack)
+  for subDir in subDirs:
+    shutil.rmtree(destDir + "/" + subDir, True)
+    shutil.move(destDir + "/" + dirToUnpack + "/" + subDir,
+                destDir + "/" + subDir)
+
+  shutil.rmtree(destDir + "/" + dirToUnpack, True)
+
+
+
+def printOptions(optionsDesc):
+  """
+  Print command line options.
+  """
+
+  print "Options:\n"
+
+  for option in optionsDesc:
+    optionUsage = "--" + option[0]
+    if option[1] != "":
+      optionUsage += "=[" + option[1] + "]"
+
+    optionDesc = option[2]
+    print "    " + optionUsage.ljust(30) + " = " + optionDesc
+
+
+
+def getCommandLineOptions():
+
+  # optionDesc = [name, value, description]
+  optionsDesc = []
+  optionsDesc.append(
+    ["nupic-core-dir",
+     "dir",
+     "(optional) Absolute path to nupic.core binary release directory"]
+  )
+  optionsDesc.append(
+    ["skip-compare-versions",
+     "",
+     "(optional) Skip nupic.core version comparison"]
+  )
+
+  # Read command line options looking for extra options
+  # For example, an user could type:
+  #   python setup.py install --nupic-core-dir="path/to/release"
+  # which will set the nupic.core release dir
+  optionsValues = dict()
+  for arg in sys.argv[:]:
+    optionFound = False
+    for option in optionsDesc:
+      name = option[0]
+      if "--" + name in arg:
+        value = None
+        hasValue = (option[1] != "")
+        if hasValue:
+          value = arg.partition("=")[2]
+
+        optionsValues[name] = value
+        sys.argv.remove(arg)
+        optionFound = True
+        break
+
+    if not optionFound:
+      if ("--help-nupic" in arg):
+        printOptions(optionsDesc)
+        sys.exit()
+
+  return optionsValues
 
 
 
-def findRequirements(repositoryDir):
+def getPlatformInfo():
+  """
+  Identify platform
+  """
+
+  if "linux" in sys.platform:
+    platform = "linux"
+  elif "darwin" in sys.platform:
+    platform = "darwin"
+  elif "win" in sys.platform:
+    platform = "win"
+  else:
+    raise Exception("Platform '%s' is unsupported!" % sys.platform)
+
+  if sys.maxsize > 2**32:
+    bitness = "64"
+  else:
+    bitness = "32"
+
+  return platform, bitness
+
+
+
+def getVersion():
+  """
+  Get version from local file.
+  """
+  with open("VERSION", "r") as versionFile:
+    return versionFile.read().strip()
+
+
+
+def generateSwigWrap(swigExecutable, swigFlags, interfaceFile):
+  """
+  Generate a swig wrap c++ file from a interface file.
```

Comment:
capitalize SWIG and C++. Also the sentence isn't quite right. Is this supposed to say something like this?

```
Generate C++ code from the specified SWIG interface file.
```